src/megaapi_impl.cpp

```diff
@@ -29551,5 +29735,505 @@ int MegaTimeZoneDetailsPrivate::getDefault() const
+    {
+        LOG_warn << "Invalid notification settings for GLOBAL.nsch";
+        return json;
+    }
+
+    json = "{";
+    if (mGlobalDND > -1 || isGlobalScheduleEnabled())
+    {        
+        json.append("\"GLOBAL\":{");
+        if (isGlobalDndEnabled())
+        {
+            json.append("\"dnd\":").append(std::to_string(mGlobalDND));
+            json.append(",");
+        }
+
+        if (isGlobalScheduleEnabled())
+        {
+            json.append("\"nsch\":{\"start\":").append(std::to_string(mGlobalScheduleStart));
+            json.append(",\"end\":").append(std::to_string(mGlobalScheduleEnd));
+            json.append(",\"tz\":\"").append(mGlobalScheduleTimezone).append("\"}");
+            json.append(",");
+        }
+    }
+
+    if (mContactsDND > -1)
+    {
+        json.append("\"PCR\":{\"dnd\":").append(std::to_string(mContactsDND)).append("}");
+        json.append(",");
+    }
+
+    if (mSharesDND > -1)
+    {
+        json.append("\"INSHARE\":{\"dnd\":").append(std::to_string(mSharesDND)).append("}");
+        json.append(",");
+    }
+
+    if (mGlobalChatsDND > -1)
+    {
+        json.append("\"CHAT\":{\"dnd\":").append(std::to_string(mGlobalChatsDND)).append("}");
+        json.append(",");
+    }
+
+    if (!mChatDND.empty() || !mChatAlwaysNotify.empty())
+    {
+        char chatid[MegaClient::CHATHANDLE * 4 / 3 + 4];
+
+        std::map<uint64_t, time_t>::const_iterator itDND;
+        for (itDND = mChatDND.begin(); itDND != mChatDND.end(); itDND++)
+        {
+            assert(isChatDndEnabled(itDND->first));
+            assert(!isChatAlwaysNotifyEnabled(itDND->first));
+
+            Base64::btoa((byte*)&(itDND->first), MegaClient::CHATHANDLE, chatid);
+            json.append("\"").append(chatid).append("\":{");
+            json.append("\"dnd\":").append(std::to_string(itDND->second)).append("}");
+            json.append(",");
+        }
+
+        std::map<uint64_t, bool>::const_iterator itAn;
+        for (itAn = mChatAlwaysNotify.begin(); itAn != mChatAlwaysNotify.end(); itAn++)
+        {
+            assert(isChatAlwaysNotifyEnabled(itAn->second));
+            assert(!isChatDndEnabled(itAn->first));
+
+            Base64::btoa((byte*)&(itAn->first), MegaClient::CHATHANDLE, chatid);
+            json.append("\"").append(chatid).append("\":{");
+            json.append("\"an\":").append("1").append("}");
+            json.append(",");
+        }
+    }
+
+    if (json.at(json.length() - 1) == ',')  // clear a tailing comma (not needed)
+    {
+        json.pop_back();
+    }
+    json.append("}");
+    return json;
+}
+
+bool MegaPushNotificationSettingsPrivate::isValid() const
+{
+    return mJsonInvalid;
+}
+
+MegaPushNotificationSettingsPrivate::~MegaPushNotificationSettingsPrivate()
+{
+
+}
+
+bool MegaPushNotificationSettingsPrivate::isGlobalEnabled() const
+{
+    return (mGlobalDND == -1 || (mGlobalDND > 0 && mGlobalDND < m_time(NULL)));
```

Comment:
Should this check consider the `nsch` (notification schedule) too? IMHO, based exclusively on the DND might be confusing. Let's discuss about it tomorrow, @alber2510